### PR TITLE
8255298: Remove SurvivorAlignmentInBytes functionality 

### PR DESCRIPTION
**Pull request by tschatzl**
Hi all,

  can I have reviews to remove the SurvivorAlignmentInBytes functionality? It has not been in use for a long time if ever, and can be removed. Searching the web also indicates that apart from the usual lists of all options and CRs it is never mentioned.

SurvivorAlignmentInBytes is an experimental option so no further process is required.

Testing: tier1-5

Thanks,
  Thomas

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8255298](https://bugs.openjdk.java.net/browse/JDK-8255298): Remove SurvivorAlignmentInBytes functionality

### Reviewers
 * [Aleksey Shipilev](https://openjdk.java.net/census#shade) (@shipilev - **Reviewer**) ⚠️ Review applies to 878cac21d74738929b31fbb29232be8d6e801d30
 * [Albert Mingkun Yang](https://openjdk.java.net/census#ayang) (@albertnetymk - Author) ⚠️ Review applies to b1407f0b1dc820e31412dae017ad27e03ed9c49c
 * [Kim Barrett](https://openjdk.java.net/census#kbarrett) (@kimbarrett - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/838/head:pull/838`
`$ git checkout pull/838`

### CODE REVIEW

File: src/hotspot/share/gc/parallel/psPromotionLAB.inline.hpp [hunk 1]
```diff
@@ -32,7 +32,7 @@
 HeapWord* PSYoungPromotionLAB::allocate(size_t size) {
   // Can't assert this, when young fills, we keep the LAB around, but flushed.
   // assert(_state != flushed, "Sanity");
-  HeapWord* obj = CollectedHeap::align_allocation_or_fail(top(), end(), SurvivorAlignmentInBytes);
+  HeapWord* obj = top();
   if (obj == NULL) {
```
Comment (shipilev) [inline src/hotspot/share/gc/parallel/psPromotionLAB.inline.hpp:36]: `obj` cannot be `NULL` now?

**Review by shipilev — CHANGES_REQUESTED** [review]
Nifty cleanup! One nit I see:

**Comment by tschatzl** [comment]
Done, thanks for catching this.

**Review by shipilev — APPROVED** [review]
These look fine to me.

**Review by albertnetymk — APPROVED** [review]

File: src/hotspot/share/gc/parallel/psPromotionLAB.inline.hpp [hunk 2]
```diff
@@ -32,17 +32,12 @@
...
   HeapWord* new_top = obj + size;
   // The 'new_top>obj' check is needed to detect overflow of obj+size.
   if (new_top > obj && new_top <= end()) {
     set_top(new_top);
-    assert(is_aligned(obj, SurvivorAlignmentInBytes) && is_object_aligned(new_top),
-           "checking alignment");
+    assert(is_object_aligned(new_top), "checking alignment");
     return obj;
   } else {
     set_top(obj);
```
Comment (kimbarrett) [inline src/hotspot/share/gc/parallel/psPromotionLAB.inline.hpp:43]: I think we don't need this set_top anymore.

**Comment by tschatzl** [comment]
@kimbarrett : fixed in new revision. Thanks!

**Review by kimbarrett — APPROVED** [review]

**Comment by tschatzl** [comment]
Thanks @kstefanj @kimbarrett @shipilev for your reviews.

/integrate

**Pull request closed without merging**